File: c++/PrototypePattern/ShallowStudent.cpp
#include "stdafx.h"
#include "ShallowStudent.h"

ShallowStudent::ShallowStudent()
{

}

ShallowStudent::ShallowStudent(const ShallowStudent &student)
{
	this->name = student.name;
	this->age = student.age;
	this->course = student.course;
}

Student *ShallowStudent::Clone()
{
	return new ShallowStudent(*this);
}

File: csharp/PrototypePattern/ShallowStudent.cs

using System;
namespace PrototypePattern
{
    /// <summary>
    /// 浅拷贝
    /// </summary>
    public class ShallowStudent : AbstractStudent
    {
        public override AbstractStudent Clone()
        {
            ShallowStudent student = new ShallowStudent();
            student.Name = base.Name;
            student.Age = base.Age;
            student.Course = base.Course;

            return student;
        }
    }
}
